### PR TITLE
Toggleable clothing changes

### PR DESCRIPTION
**Pull request by carteblanche4me**
## About the PR

The slot where to equip a toggleable clothing piece can and should be specified in prototypes. Already existing toggleable clothing slots are specified. The toggle action ID, name, and description are logically changed.

## Why / Balance

The PR creates new opportunities for prototype creators and excludes hardcode.

## Technical details

Changed hardcode to ViewVariables used for other clothing components' parameters.
The changes have been tested with different clothing parts by trying to unequip them yourself, by other players, and by turning a player into a zombie.

If an article of clothing that has been removed manually has an in-hand model, it is displayed in the hand for a moment (count is in milliseconds).

## Media

image
_The custom captain spacesuit prototype with toggling gloves instead of a helmet for testing._
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: carteblanche4me
- tweak: Hardsuits could toggle on/off not only helmets but different pieces.

### CODE REVIEW

File: Resources/Locale/en-US/actions/actions/hardsuit.ftl [hunk 1]
```diff
@@ -1,2 +1,2 @@
-action-name-hardsuit = Toggle Helmet
-action-description-hardsuit = Remember to equip the helmet before flinging yourself into space.
\ No newline at end of file
+action-name-hardsuit = Toggle Suite Piece
```
Comment (metalgearsloth) [inline Resources/Locale/en-US/actions/actions/hardsuit.ftl:1]: suit

File: Content.Shared/Clothing/Components/ToggleableClothingComponent.cs [hunk 2]
```diff
@@ -33,8 +33,9 @@ public sealed partial class ToggleableClothingComponent : Component
     /// <summary>
     ///     The inventory slot that the clothing is equipped to.
     /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
     [DataField("slot")]
-    public string Slot = "head";
+    public string Slot;
```
Comment (metalgearsloth) [inline Content.Shared/Clothing/Components/ToggleableClothingComponent.cs:38]: Should this just use slotflags or the other enum.

Comment (carteblanche4me) [reply Content.Shared/Clothing/Components/ToggleableClothingComponent.cs:38]: Right, enum is the perfect solution here, but I don't know all the slot names and wasn't able to find them. Any advice is welcome.

File: Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml [hunk 3]
```diff
@@ -26,6 +26,7 @@
     sprintModifier: 0.80
   - type: ToggleableClothing
     clothingPrototype: ClothingHeadHelmetHardsuitBasic
+    slot: head
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml:29]: if you add that to parent you don't need to use it again in prototypes that uses that as parent

File: Resources/Prototypes/Entities/Clothing/OuterClothing/base_clothingouter.yml [hunk 4]
```diff
@@ -68,6 +68,7 @@
         Heat: 0.90
         Radiation: 0.25
   - type: ToggleableClothing
+    slot: head
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Clothing/OuterClothing/base_clothingouter.yml:71]: oh, you can keep it only here

Comment (carteblanche4me) [reply Resources/Prototypes/Entities/Clothing/OuterClothing/base_clothingouter.yml:71]: You're missing that there are a few cloaks and an EVA softsuit that use toggleable clothing. I get your point regarding parental prototypes, but I think it should be everywhere for better understanding. I was even thinking about not including it in the parent hardsuit prototype, but decided to keep it for already existing custom prototypes to be operable.

File: Content.Shared/Clothing/Components/ToggleableClothingComponent.cs (hunk 2 above)
Comment (metalgearsloth) [reply Content.Shared/Clothing/Components/ToggleableClothingComponent.cs:38]: Okay yeah SlotFlags is the one.

Comment (carteblanche4me) [reply Content.Shared/Clothing/Components/ToggleableClothingComponent.cs]: I'm blind, it's literally SlotFlags, now it's done.

File: Resources/Prototypes/Entities/Clothing/OuterClothing/base_clothingouter.yml (hunk 4 above)
Comment (carteblanche4me) [reply Resources/Prototypes/Entities/Clothing/OuterClothing/base_clothingouter.yml:71]: I have reconsidered your comment, you're right. Hardsuits can't really exist without helmets, so slot specification is kept in the parental prototype only.

**Comment by JoeHammad1844** [comment]
demonstration, please

**Comment by carteblanche4me** [comment]
> demonstration, please

Of what? Now it is possible to create a prototype that can toggle different clothing pieces, not just the head slot. I'm not suggesting any examples yet. There is the testing variant of the cap suit with toggleable gloves in the screenshot.
Let me know if you want something specific, and we'll try to figure out if I can record that for you.

File: Content.Shared/Clothing/Components/ToggleableClothingComponent.cs (hunk 2 above)
Comment (metalgearsloth) [reply Content.Shared/Clothing/Components/ToggleableClothingComponent.cs]: Okay I ended up not liking this after opening up my IDE and looking further so I just reverted it on my end.

**Comment by metalgearsloth** [comment]
@carteblanche4me you might also want to use branches and not dev on master too.

**Pull request merged**